src/SIL.XForge.Scripture/ClientApp/src/app/core/sfproject.service.ts

```diff
@@ -2,19 +2,18 @@ import { Injectable } from '@angular/core';
 
 import { JSONAPIService } from '@xforge-common/jsonapi.service';
 import { OptimisticResourceService } from '@xforge-common/optimistic-resource.service';
-import { record } from '@xforge-common/resource-utils';
-import { SFProject, SFPROJECT, SFProjectAttributes } from '../shared/models/sfproject';
+import { SFProject } from '../shared/models/sfproject';
 
 @Injectable({
   providedIn: 'root'
 })
-export class SFProjectService extends OptimisticResourceService<SFProject, SFProjectAttributes> {
+export class SFProjectService extends OptimisticResourceService<SFProject> {
 
   constructor(jsonApiService: JSONAPIService) {
-    super(jsonApiService, SFPROJECT);
+    super(jsonApiService, SFProject.TYPE);
   }
 
-  onlineCreate(resource: Partial<SFProject>): Promise<string> {
-    return this.jsonApiService.create(record(this.type, resource), true, true);
+  onlineCreate(resource: SFProject): Promise<string> {
```

Comment:
Why a promise instead of an observable here?

Reply:
We decided that we would use observables for queries and promises for actions and transforms. Observables feel like an unnecessary complication for actions.